src/editor/lexercpp.cpp

```diff
@@ -1,6 +1,212 @@
+}
+
+bool LexerCPP::isNumber(QString lexem)
+{
+    QRegExp rx("[0-9]+");
+    return rx.exactMatch(lexem);
+}
+
+bool LexerCPP::isFloatNumber(QString lexem)
+{
+    QRegExp rx("[0-9]+\\.[0-9]+");
+    return rx.exactMatch(lexem);
+}
+
+bool LexerCPP::isOperator(QString lexem)
+{
+    return OPERATORS.contains(lexem);
+}
+
+bool LexerCPP::isSymbolLiteral(QString lexem)
+{
+    QRegExp rx("'.'|'\.+'");
+    return rx.exactMatch(lexem);
+}
+
+bool LexerCPP::isStringLiteral(QString lexem)
+{
+    QRegExp rx("\".*\"");
+    return rx.exactMatch(lexem);
+}
+
+bool LexerCPP::isOneLineComment(QString lexem)
+{
+    QRegExp rx("//.*");
+    return rx.exactMatch(lexem);
+}
+
+bool LexerCPP::isBlockComments(QString lexem)
+{
+    QRegExp rx("/\\*[\\s\\S]*\\*/");
+    return rx.exactMatch(lexem);
+}
+
+bool LexerCPP::isSpace(QChar sym)
+{
+    return SPACES.contains(sym);
+}
+
+bool LexerCPP::isAlpha(QChar sym)
+{
+    QRegExp rx("[A-Za-z]");
+    return rx.exactMatch(QString(sym));
+}
+
+bool LexerCPP::isDigit(QChar sym)
+{
+    QRegExp rx("[0-9]");
+    return rx.exactMatch(QString(sym));
+}
+
+void LexerCPP::addLexem()
+{
+    --it;
+    tokens.append(Token(QString(current_lexem), States(state), it - symbolCount - current_lexem.size(), it - symbolCount, current_line));
+    current_lexem.clear();
+    state = ST;
+}
+
+void LexerCPP::changeState(States st, QChar sym)
+{
+    state = st;
+    current_lexem += sym;
+}
+
+void LexerCPP::clear()
+{
+    tokens.clear();
+    current_lexem.clear();
+    state = ST;
+    symbolCount = 0;
+    current_line = 0;
+}
+
+void LexerCPP::lexicalAnalysis(QString code)
+{
+    it = 0;
+    QChar sym = 0;
+    while(code.size() - it)
+    {
+        sym = code[it];
+        ++it;
+        switch(state)
+        {
+            case ST:
+                if(isSpace(sym)) {
+                    if(sym == '\n')
+                    {
+                        symbolCount = it;
+                        ++current_line;
+                    }
+                    break;
+                }
+                if(isAlpha(sym))
+                    changeState(ID, sym);
+                else if(isDigit(sym))
+                    changeState(NUM, sym);
+                else if(isOperator(QString(sym)))
+                    changeState(OPER, sym);
+                else if(sym == '\'' || sym == '"')
+                    changeState(LIT, sym);
+            break;
+
+            case ID:
+                if(isSpace(sym) || isOperator((QString)sym) || sym == '\"' || sym == '\'')
+                    addLexem();
+                else if(isIdentifier(current_lexem + sym))
+                {
+                    current_lexem += sym;
+                    if(isKeyword(current_lexem))
+                        state = KW;
+                }
+                else
+                    state = UNDEF;
+            break;
+
+            case KW:
+                if(isSpace(sym) || isOperator((QString)sym) || sym == '\"' || sym == '\'')
```

Comment:
It will be better to create some interface.  You can create map<state, IStateHandler> then you should  implement concrete  StateHandler for each state